internal/agent/agent.go

```diff
@@ -14,14 +15,21 @@ import (
 	"go.uber.org/zap"
 )
 
+// Agent represents a metrics agent that collects and reports metrics.
 type Agent struct {
-	serverAddr     string
-	pollInterval   time.Duration
+	// ServerAddr is the address of the server.
+	serverAddr string
```

Comment:
Структуры принято комментировать так.
```suggestion
	serverAddr string // ServerAddr is the address of the server.
```